conda/models/environment.py

```diff
@@ -358,3 +393,101 @@ def merge(cls, *environments):
             variables=variables,
             virtual_packages=virtual_packages,
         )
+
+    @classmethod
+    def from_prefix(
+        cls,
+        prefix: str,
+        name: str,
+        platform: str,
+        *,
+        from_history: bool = False,
+        no_builds: bool = False,
+        ignore_channels: bool = False,
+        channels: list[str] | None = None,
+        override_channels: bool = False,
+    ) -> Environment:
+        """
+        Create an Environment model from an existing conda prefix.
+
+        This method analyzes an installed conda environment and creates
+        an Environment model that can be used for exporting or other operations.
+
+        :param prefix: Path to the conda environment prefix
+        :param name: Name for the environment
+        :param platform: Target platform (e.g., 'linux-64', 'osx-64')
+        :param from_history: Use explicit specs from history instead of installed packages
+        :param no_builds: Exclude build strings from package specs
+        :param ignore_channels: Don't include channel information in package specs
+        :param channels: Additional channels to include in environment config
+        :param override_channels: Whether to override default channels
+        :return: Environment model representing the prefix
+        """
+        prefix_data = PrefixData(prefix)
+
+        # Build requested packages and explicit packages
+        requested_packages = []
+        explicit_packages = []
+
+        # Always populate explicit_packages from prefix data (all installed packages)
+        for prefix_record in prefix_data.iter_records():
+            explicit_packages.append(prefix_record)
+
+        # Populate requested_packages from history if available and requested
+        if from_history:
+            history = History(prefix)
+            spec_map = history.get_requested_specs_map()
+            requested_packages = list(spec_map.values())
+        else:
+            # Fall back to creating MatchSpecs from all installed packages
+            for prefix_record in prefix_data.iter_records():
+                if no_builds:
+                    spec_str = f"{prefix_record.name}=={prefix_record.version}"
+                else:
+                    spec_str = f"{prefix_record.name}=={prefix_record.version}={prefix_record.build}"
```

Comment:
We have added this stringification a number of times in this PR, should this be a property added to the `PackageRecord`?